x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapRealmTests.java

```diff
@@ -142,6 +142,98 @@ public void testAuthenticateSubTreeGroupSearch() throws Exception {
+        ldap.authenticate(new UsernamePasswordToken("John Samuel", new SecureString(PASSWORD)), future);
+        final AuthenticationResult<User> result = future.actionGet();
+        assertThat(result.getStatus(), is(AuthenticationResult.Status.SUCCESS));
+        User user = result.getValue();
+        assertThat(user, notNullValue());
+        assertThat(user.email(), nullValue());
+        assertThat(user.fullName(), equalTo("John Samuel"));
+    }
+
+    public void testAuthenticateMapFullNameAndEmailMetadata() throws Exception {
+        String groupSearchBase = "o=sevenSeas";
+        Settings settings = Settings.builder()
+            .put(defaultGlobalSettings)
+            .put(buildLdapSettings(ldapUrls(), VALID_USER_TEMPLATE, groupSearchBase, LdapSearchScope.SUB_TREE))
+            .put(getFullSettingKey(REALM_IDENTIFIER, RealmSettings.ORDER_SETTING), 0)
+            .put(getFullSettingKey(REALM_IDENTIFIER, LdapMetadataResolverSettings.FULL_NAME_SETTING), "description")
+            .put(getFullSettingKey(REALM_IDENTIFIER, LdapMetadataResolverSettings.EMAIL_SETTING), "uid")
+            .build();
+        RealmConfig config = getRealmConfig(REALM_IDENTIFIER, settings);
+        SessionFactory ldapFactory = LdapRealm.sessionFactory(config, sslService, threadPool);
+        LdapRealm ldap = new LdapRealm(config, ldapFactory, buildGroupAsRoleMapper(resourceWatcherService), threadPool);
+        ldap.initialize(Collections.singleton(ldap), licenseState);
+
+        PlainActionFuture<AuthenticationResult<User>> future = new PlainActionFuture<>();
+        ldap.authenticate(new UsernamePasswordToken("William Cole", new SecureString(PASSWORD)), future);
+        final AuthenticationResult<User> result = future.actionGet();
+        assertThat(result.getStatus(), is(AuthenticationResult.Status.SUCCESS));
+        User user = result.getValue();
+        assertThat(user, notNullValue());
+        assertThat(user.email(), equalTo("wcole@royalnavy.mod.uk"));
+        assertThat(user.fullName(), equalTo("William Mark Cole"));
+    }
+
+    public void testAuthenticateMapNonExistingMetadata() throws Exception {
+        String groupSearchBase = "o=sevenSeas";
+        Settings settings = Settings.builder()
+            .put(defaultGlobalSettings)
+            .put(buildLdapSettings(ldapUrls(), VALID_USER_TEMPLATE, groupSearchBase, LdapSearchScope.SUB_TREE))
+            .put(getFullSettingKey(REALM_IDENTIFIER, RealmSettings.ORDER_SETTING), 0)
+            .put(getFullSettingKey(REALM_IDENTIFIER, LdapMetadataResolverSettings.FULL_NAME_SETTING), "thisdoesnotexist")
+            .put(getFullSettingKey(REALM_IDENTIFIER, LdapMetadataResolverSettings.EMAIL_SETTING), "thisdoesnotexist")
+            .build();
+        RealmConfig config = getRealmConfig(REALM_IDENTIFIER, settings);
+        SessionFactory ldapFactory = LdapRealm.sessionFactory(config, sslService, threadPool);
```

Comment:
I really think there's too much test code overhead to only check that the new settings work correctly.
If you look at the other test methods, they deal with more elaborate concepts (like the types of group search, realm level caching, delegate authz).
Besides the amount of test code overhead, another consequence of this is that the test only asserts that _some_ particular LDAP realm configuration works with the new settings.

I think it's fine to have here **one** test method that deals exclusively with the new settings, but randomize some of the other realm settings (e.g. have multiple realms). Alternatively I think it's also fine to (randomly) add the new settings to some of the existing test methods that deal with more elaborate concepts (I would also check that the settings work in the delegated authorization context , i.e. `testDelegatedAuthorization`).

Reply:
Yes, great point. Will change this. Thanks!